Visual improvements

pmaas opened the pull request:
Added some style improvements to integrate the color picker a  little bit better into Neos:

**Before / After**
<img width="315" alt="Bildschirmfoto 2019-07-17 um 12 44 57" src="https://user-images.githubusercontent.com/2778669/61369609-c081cd80-a890-11e9-9ebc-669754eb6180.png"><img width="316" alt="Bildschirmfoto 2019-07-17 um 12 28 28" src="https://user-images.githubusercontent.com/2778669/61368908-32f1ae00-a88f-11e9-9362-54544a202e34.png">

I needed to create a custom picker based on the sketch color picker, because I wasn't able to style all components by the `style`-prop. But thereby it is also possible to integrate the reset button into the picker.

Sebobo commented:
Wow! Thanks!

I'll review and merge it asap

Sebobo commented on Resources/Private/Scripts/ColorPickerEditor/src/NeosColorPicker/Alpha.js:
```diff
@@ -0,0 +1,113 @@
+
+import Checkboard from 'react-color/lib/components/common/Checkboard'
+
+export class Alpha extends (PureComponent || Component) {
```
Would be good to define the possible proptypes here

pmaas replied:
Fixed

Sebobo commented on Resources/Private/Scripts/ColorPickerEditor/src/NeosColorPicker/ColorPicker.js:
```diff
@@ -0,0 +1,169 @@
+import {IconButton} from '@neos-project/react-ui-components';
+
+export const ColorPicker = ({ width, rgb, hex, hsv, hsl, onChange, onReset, onSwatchHover,
+  disableAlpha, presetColors, renderers, styles: passedStyles = {}, className = '' }) => {
```
Is `disableAlpha` forwarded to the fields? I cannot see where and PHPStorm shows it's unused.

pmaas replied:
Maybe we can make this a future feature. Until this I removed the variable.

Sebobo commented on Resources/Private/Scripts/ColorPickerEditor/src/NeosColorPicker/Fields.js:
```diff
@@ -0,0 +1,147 @@
+/* eslint-disable no-param-reassign */
+
+import React from 'react'
+import reactCSS, { focus } from 'reactcss'
```
`focus` does not seem to be used

pmaas replied:
Is now removed.

Sebobo commented on Resources/Private/Scripts/ColorPickerEditor/src/NeosColorPicker/Hue.js:
```diff
@@ -0,0 +1,93 @@
+	}
+	
+	handleChange = (e) => {
+		const change = hue.calculateChange(e, this.props.direction, this.props.hsl, this.container)
```
I get an error `Cannot read property 'h' of undefined` here when using the `hsla` mode and selecting a color from the rgb and alpha bars. Selections from the other fields work fine.

pmaas replied:
Sorry for the bad testing. I added some last minute style changes and didn't check the functionality afterwards.

Sebobo commented on Resources/Private/Scripts/ColorPickerEditor/src/ColorPickerEditor.js:
```diff
@@ -1,15 +1,15 @@
 import PropTypes from 'prop-types';
-import {SketchPicker} from 'react-color';
+import NeosColorPicker from './NeosColorPicker/ColorPicker';
 import {IconButton} from '@neos-project/react-ui-components';
```
`IconButton` is not used anymore

pmaas replied:
Is now removed.

pmaas commented:
Hey @Sebobo, thanks for the review! And sorry for the late reply, I've been on vacation last week, but I try to do the fixes asap.

pmaas commented:
Hey @Sebobo, I added the `.eslintrc` from Neos.Ui as recommended by you and fixed the hue/alpha slider bugs. Would be awesome if you find some time to check the changes. Thanks!

Sebobo commented:
Looks good. I fixed some minor details.

Can you define the missing `propTypes` in Alpha & Hue? Then we're done I think :)

pmaas commented:
> Can you define the missing propTypes in Alpha & Hue? Then we're done I think :)
Done.
